### PR TITLE
Make sure the graph death test has the `_DeathTest` suffix

### LINKED ISSUE #7257: Nightly build failures, OpenMP backend in debug mode, openmp_graph.can_instantiate_only_once triggers Subprocess aborted
**Describe the bug**

Some nightly builds with OpenMP enabled with debug mode are failing in `openmp_graph.can_instantiate_only_once triggers` with `Subprocess aborted`

This is occurring for example with intel/19.0.5 icpc OpenMP debug builds, and cuda/11.2.2+gcc/8.5.0 with Cuda+OpenMP debug builds

**Changes**

    Fix misspelled cmake variable, for some systems leading to not compiling all tests ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#6e94ed0fd872c8fd9865526200cd7b561e17d752) / [githubweb](https://github.com/kokkos/kokkos/commit/6e94ed0fd872c8fd9865526200cd7b561e17d752))
    Fix overlooked naming test in #7222 ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#46d2349c1b2b5625d8205ea435f291a37bed9894) / [githubweb](https://github.com/kokkos/kokkos/commit/46d2349c1b2b5625d8205ea435f291a37bed9894))
    Use macros to protect use of SharedSpace ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#506f625bcb318feb65b290fda6938464343f5be2) / [githubweb](https://github.com/kokkos/kokkos/commit/506f625bcb318feb65b290fda6938464343f5be2))
    core(graph): promote `instantiate` to public API ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#9f71f2048b79b6944ca27665af6083ddc9625023) / [githubweb](https://github.com/kokkos/kokkos/commit/9f71f2048b79b6944ca27665af6083ddc9625023))
    Enforce modernize-type-traits (#7227) ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#23c1fab67e1f0fe882ddaa1d2acf275dc52ae35c) / [githubweb](https://github.com/kokkos/kokkos/commit/23c1fab67e1f0fe882ddaa1d2acf275dc52ae35c))
    typo: kokkkos -> kokkos ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#e5312197bc5d3ffc6ce8be613679e805770dde92) / [githubweb](https://github.com/kokkos/kokkos/commit/e5312197bc5d3ffc6ce8be613679e805770dde92))
    core(view): aligning `HostMirror` and `host_mirror_type` ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#030d83b3259a669ef61aecfbbdd4b8139bb6efab) / [githubweb](https://github.com/kokkos/kokkos/commit/030d83b3259a669ef61aecfbbdd4b8139bb6efab))
    Add support for CUDA unified memory architectures i.e. Grace Hopper (#6823) ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#b367a71b8e9f7d4f1c89231f5b4c6ef09ee77626) / [githubweb](https://github.com/kokkos/kokkos/commit/b367a71b8e9f7d4f1c89231f5b4c6ef09ee77626))
    Using CUDA limits to set extents for blocks,grids (#7235) ([details](https://jenkins-son.sandia.gov/job/kokkos_osrhel7_intel_1905-dbg/1280/changes#1f99b1d3851c780aa18829ff4bf2b9b61bb9df21) / [githubweb](https://github.com/kokkos/kokkos/commit/1f99b1d3851c780aa18829ff4bf2b9b61bb9df21))

**Please include the following for a minimal reproducer**

**Reproducer** (intel/19 shown here):
```bash
module load sems-cmake sems-intel/19.0.5

$KOKKOS_PATH/generate_makefile.bash --with-devices=OpenMP  --compiler=icpc --cxxflags="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -diag-disable=1011,-diag-disable=869" --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --no-examples   --with-options=enable_large_mem_tests --debug
```

### PR DESCRIPTION
**Pull request by dalg24**
Fix #7257

### CODE REVIEW

File: core/unit_test/TestGraph.hpp [hunk 1]
```diff
@@ -143,12 +143,23 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), instantiate_and_submit_once) {
   ASSERT_EQ(0, bugs_host());
 }
 
+// FIXME death tests an fixtures
```
Comment (masterleinad) [inline core/unit_test/TestGraph.hpp:146]:
```suggestion
// FIXME death tests and fixtures
```

**Review by masterleinad — APPROVED** [review]

**Review by romintomasetti — APPROVED** [review]

**Comment by dalg24** [comment]
@ndellingwood please confirm that this resolves the issue you reported from the nightly builds

**Review by ndellingwood — APPROVED** [review]
Test now passes in both failing cases I posted, thanks @dalg24 !

**Comment by dalg24** [comment]
Ignoring warning treated was an error in one of the CUDA builds.  Everything else passed.

**Pull request merged**